cmd/cluster-agent/app/app.go

```diff
@@ -184,80 +185,81 @@ func start(cmd *cobra.Command, args []string) error {
 
 	log.Infof("Datadog Cluster Agent is now running.")
 
-	apiCl, err := apiserver.GetAPIClient() // make sure we can connect to the apiserver
+	retrierCtx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
I think you can just give `context.Background`, no need to hardcode timeout a second time here.
You can trust `WaitForAPIClient` to return after 5mins in any case.

Reply:
I don't see a reason why `WaitForAPIClient` takes a context argument then

Reply:
made the change you suggested anyway, we can consider changing WaitForAPIClient's signature in the future

Reply:
It's because you might have a shorter context/deadline. But if you're willing to wait until permafail `context.Background` is enough.

Reply:
I'm going to merge it, any other comments?